server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -149,6 +149,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion AGGS_EXCLUDED_DELETED_DOCS = def(8_609_00_0);
     public static final TransportVersion ESQL_SERIALIZE_BIG_ARRAY = def(8_610_00_0);
     public static final TransportVersion AUTO_SHARDING_ROLLOVER_CONDITION = def(8_611_00_0);
+    public static final TransportVersion FAILURE_STORE_ROLLOVER = def(8_612_00_0);
```

Comment:
Yep, your imagination is correct. I did think about renaming this, but I thought it was fine as I'll add the other changes soon (i.e. the same release version). Would you suggest having two different version names (one for failure store generation and one for rollover), or renaming this to failure store generation and using that in the rollover changes as well? The rollover changes require a transport version because I've added a field to `RolloverRequest`.

Reply:
I'm more concerned about these changes being picked up in cloud deploys where the versions there are not always tied to release versions, hence why the `TransportVersion` class replaced the old `Version` class in most checks. There will likely be a number of commits between this change and the next and we'll need to ensure mixed cluster compatibility between them. It seems a little excessive, yeah, but it's the "right way to do it" for now.

> Would you suggest having two different version names (one for failure store generation and one for rollover), or renaming this to failure store generation and using that in the rollover changes as well?

I think we should do one for failure store generation and one for rollover. These are technically discreet changes which means they'll need to be tracked in `TransportVersion` separately.